internal/pkg/cli/svc_init.go

```diff
@@ -180,11 +182,8 @@ func newInitSvcOpts(vars initSvcVars) (*initSvcOpts, error) {
 	return opts, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error for any invalid optional flags.
 func (o *initSvcOpts) Validate() error {
-	if o.appName == "" {
-		return errNoAppInWorkspace
-	}
 	if o.wkldType != "" {
```

Comment:
> Do we consider --subscribe-topics required for worker service?

It's an optional flag actually because a worker service could subscribe no service.